### ISSUE TITLE
ignored-modules should turn no-name-in-module-off

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#223]
Originally reported by: **Antony Lee (BitBucket: [anntzer](http://bitbucket.org/anntzer), GitHub: @anntzer?)**

---

I believe no-name-in-module is basically the same as no-member (for modules), except that the former triggers at "from ... import" statements.  Modules in the "ignore-modules" list should thus be ignored for no-name-in-module too.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/223

**Comment by glennmatthews** [comment]
This seems to still be an issue in 1.5.6. Am I missing something?

**Comment by bukzor** [comment]
It works for me. Please give us a reproduction.

http://sscce.org

**Comment by glennmatthews** [comment]
User error on my part (duplicate ignored-modules key in .pylintrc). Looks like this kind of user error will be caught in future by Pylint due to moving to use the newer configparser module (#828) which rejects duplicates by default. Sorry for the noise!

**Comment by parkan** [comment]
1.6.0 seems to have broken the use of `disable=no-name-in-module`:

```
E:  5, 0: Bad option value 'no-name-in-module' (bad-option-value)
```

This is the only relevant issue I could find from the release notes; is no-name-in-module now superseded by ignored-modules?

**Comment by PCManticore** [comment]
I cannot reproduce this, it works fine for me, can you provide more details, please?

**Comment by PCManticore** [comment]
If you can provide the astroid version, the snippet of code for which this reproduces and how you called pylint, that would be helpful.

**Comment by parkan** [comment]
```
[pylint-repro]% cat test/test.py                              (arkadiy@helium:/tmp/pylint-repro)
import doesnotexist  # pylint: disable=no-name-in-module
[pylint-repro]% pylint --errors-only test                     (arkadiy@helium:/tmp/pylint-repro)
Problem importing module variables.py: No module named functools_lru_cache
Problem importing module variables.pyc: No module named functools_lru_cache
No config file found, using default configuration
************* Module test.test
E:  1, 0: Bad option value 'no-name-in-module' (bad-option-value)
[pylint-repro]% python --version                              (arkadiy@helium:/tmp/pylint-repro)
Python 2.7.9
[pylint-repro]% pylint --version                              (arkadiy@helium:/tmp/pylint-repro)
Problem importing module variables.py: No module named functools_lru_cache
Problem importing module variables.pyc: No module named functools_lru_cache
No config file found, using default configuration
pylint 1.6.0,
astroid 1.4.7
Python 2.7.9 (default, Jan  7 2015, 11:49:12)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.56)]
```

**Comment by parkan** [comment]
Also happens in our travis build here https://travis-ci.org/mediachain/mediachain-client/builds/143118789

**Comment by PCManticore** [comment]
I figured out the problem and I will issue a release shortly.

**Comment by parkan** [comment]
@PCManticore thanks so much for the quick response!

**Comment by PCManticore** [comment]
@parkan can you check with pylint 1.6.1? It should work not as expected.

**Comment by parkan** [comment]
Regression fixed 👍

**Comment by yagi-takeru** [comment]
seems to have popped back up as of 1.7.2

```file: [redacted]
severity: 'Error'
message: 'E0401:Unable to import 'six.moves.urllib.parse''
at: '9,1'
source: 'pylint'```

**Comment by hutchk** [comment]
Using 1.7.4 still seeing this problem.

**Comment by anderson-dan-w** [comment]
Was seeing this error in `pylint=1.6.5`, `astroid=1.4.9`, but found that, in my case:

```python
from six.moves.urllib.parse import urlencode
```
the fix was:

`ignored-modules='six.moves.urllib,six.moves.urllib.parse'`
